api/api.py

```diff
@@ -13,29 +13,34 @@ def hello_json():
 
 @api_bp.route('/version', methods=['GET'])
 def get_api_version():
-    return jsonify({'Json sagt': API_VERSION })
-
-
-@api_bp.route(API_VERSION + '/users/<uid>', methods=['GET'])
-def get_current_user(uid):
-    user = find_user(username=uid)
-    return jsonify(user)
-
-
-@api_bp.route(API_VERSION + '/webapps', methods=['GET'])
-def get_all_webapps():
-    return jsonify(mock.webapps)
-
-
-@api_bp.route(API_VERSION + '/statistics/<stat>', methods=['GET'])
-def get_cluster_statistics(stat):
-    statistics = mock.metrics
-    '''
-    iterate through the list of metrics, remove whitespaces in key and convert key to lowerspace to match with url param
-    '''
-    for metric in statistics:
-        if next(iter(metric)).lower().replace(' ', '') == stat:
-            return jsonify(metric)
-    return jsonify({'metric': None})
-
-
+    return jsonify({'Json sagt': API_VERSION}), status.HTTP_200_OK
+
+
+@api_bp.route(f'{API_VERSION}/cluster/<info>', methods=['GET'])
+def get_cluster_info(info):
+    cluster_info = mock.cluster_info
+    for key, value in cluster_info.items():
+        if info == key:
+            response = jsonify({info: value})
+            return response, status.HTTP_200_OK
+    return jsonify(status.HTTP_404_NOT_FOUND)
+
+
+@api_bp.route(f'{API_VERSION}/webapps/<id>', methods=['GET'])
+def get_webapp(id):
+    webapps = mock.webapps
+    webapp = next((webapp for webapp in webapps if webapp['id'] == id), None)
+    if webapp:
+        response = jsonify({"link_name": webapp.get('link_name')}, {"link_url": webapp.get('link_url')})
+        return response, status.HTTP_200_OK
+    return jsonify(status.HTTP_400_BAD_REQUEST)
+
+
+@api_bp.route(f'{API_VERSION}/groups/<id>', methods=['GET'])
+def get_group(id):
+    groups = mock.groups
+    group = next((group for group in groups if group['id'] == id), None)
```

Comment:
group group group group group ..... (chicken is coming)

Reply:
I appreciate your qualified feedback.